modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3349,10 +3353,10 @@ def n_ary_op(
 
         return self.__constructor__(
             new_frame,
-            joined_index,
-            joined_columns,
-            row_lengths,
-            column_widths,
+            joined_index if labels == "replace" else None,
+            joined_columns if labels == "replace" else None,
+            row_lengths if labels == "replace" else None,
+            column_widths if labels == "replace" else None,
```

Comment:
Otherwise, there will be a mismatch between the internal and external indexes of the object after the operation but before extracting the scalar.